### PR TITLE
STL.natvis: add specializations for `shared_ptr<void>`, `weak_ptr<void>`, and `unique_ptr<void, deleter>`

### LINKED ISSUE #2688: STL.natvis: shared_ptr<void> outputs an error
**Describe the bug**

When inspecting an `std::shared_ptr<void>` in the debugger, it outputs an error. This is due to [the following line](https://github.com/microsoft/STL/blob/main/stl/debugger/STL.natvis#L395) in the natvis file:
```
<DisplayString IncludeView="ptr" Condition="_Ptr != 0">{*_Ptr}</DisplayString>
```
In particular, `std::shared_ptr<void>` matches the condition (`_Ptr != 0`) but still we can't dereference it (`*_Ptr`).
The error also is pretty clear about the reasons:
> Natvis: C:\<path>\stl.natvis(395,63): Error: expression must be a pointer to a complete object type
>    Error while evaluating '*_Ptr' in the context of type '<executable>.exe!std::shared_ptr<void>'.

**Command-line test case**

N/A (that's a natvis issue).

**Expected behavior**

Inspecting an `std::shared_ptr<void>` doesn't output any errors.

**STL version**

VS Community 2022 v17.1.6

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by fsb4000**
Fixes #2688 

It works, but perhaps there is an option without the duplication...

### CODE REVIEW

**Review by strega-nil-ms — APPROVED** [review]
LGTM, thanks!

**Comment by CaseyCarter** [comment]
> It works, but perhaps there is an option without the duplication...

We could tag the problematic part of the original visualizer Optional, but the result would be less than ideal. I don't know how to otherwise avoid this duplication.

**Review by CaseyCarter — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]

File: stl/debugger/STL.natvis [hunk 1]
```diff
@@ -288,6 +288,16 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
   </Type>
 
   <!-- VC 2015 -->
+   <Type Name="std::unique_ptr&lt;void, *&gt;">
```
Comment (CaseyCarter) [inline stl/debugger/STL.natvis:291]: I've pulled these changes into internal PR [VS-400483](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/400483). Please notify me of any changes to this file.

Comment (StephanTLavavej) [reply stl/debugger/STL.natvis:291]: I'll add this to the MSVC-PR mirror too.

Comment (CaseyCarter) [reply stl/debugger/STL.natvis:291]: May as well. I don't plan to include the "big" PR. I won't have time to review it today, `expected` is taking an ... unexpectedly long time to review. 😎

**Comment by StephanTLavavej** [comment]
Thanks for fixing these visualizers! :smile_cat: :eye: :tada:

**Pull request merged**